File: crates/infergen-core/templates/sveltekit/snippet.ts
// After `infergen generate`, import your typed SDK:
import { track } from "$lib/infergen.generated";

// In +layout.ts or +page.ts:
export const load = ({ url }) => {
  track.pageViewed({ page_path: url.pathname });
};

// In a +page.server.ts action:
export const actions = {
  default: async ({ request }) => {
    track.formActionInvoked({ action_name: "default" });
  },
};

File: crates/infergen-core/templates/rails/snippet.rb
# After `infergen generate`, require your typed SDK:
require_relative "infergen/generated"

# In ApplicationController:
class ApplicationController < ActionController::Base
  before_action :track_action

  private

  def track_action
    Infergen::Track.controller_action_called(
      controller: controller_name,
      action: action_name
    )
  end
end

# After Devise sign-in:
def after_sign_in_path_for(resource)
  Infergen::Track.user_signed_in(user_id: current_user.id.to_s)
  root_path
end
